lib/auth/auth_with_roles.go

```diff
@@ -833,12 +851,23 @@ func (a *ServerWithRoles) checkAccessToResource(resource types.Resource) error {
 		return a.checkAccessToApp(r.GetApp())
 	case types.DatabaseServer:
 		return a.checkAccessToDatabase(r.GetDatabase())
+	case types.Server:
+		if resource.GetKind() == types.KindKubeService {
+			// KubeServices are present inside the type Server. We need to
+			// filter out the ones that the user doesn't have access to. This
+			// function won't return an AccessDenied because we're not checking
+			// access to the Server itself.
+			return a.filterKubeServices(r)
+		}
+
+		return a.checkAccessToNode(r)
```

Comment:
Since permissions can vary based on resource kind here, we should err out on unexpected resource kinds rather than assuming that the resource is a node if it isn't a kube service (help catch future bugs).